### PR TITLE
Adds date and code block instructions to AssistantAgent prompt. 

### PR DESCRIPTION
**Pull request by afourney**
## Why are these changes needed?

There are several opportunities to improve the default AssistantAgent system message to:

- Make the agent aware of the current date. (#428)
- Provide more explicit instructions for formatting code (more important for small LLMs)

## Related issue number

#428 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by afourney** [comment]
Initial testing with a reduced 26-problem version of HumanEval shows promise. No appreciable difference for GPT-4 (green) or GPT-3.5-turbo-16k (red), but sizeable boosts to older GPT3.5-turbo-0301 (blue) and Zephyr-7B (purple).

Gains mainly come from adherence to Markdown code block formatting, aiding detection and execution by the UserProxyAgent.

image

**Comment by gagb** [comment]
@afourney shall we revisit this post results with GAIA?

**Comment by afourney** [comment]
@gagb I don't expect it to impact GAIA much because GAIA basically requires GPT-4 (it's too hard for other models), and GPT-4 is ok with this.

This change is mostly for the benefit of weaker models like GPT-3.5

File: autogen/agentchat/assistant_agent.py [hunk 1]
```diff
@@ -14,18 +14,22 @@ class AssistantAgent(ConversableAgent):
...
 
-    DEFAULT_SYSTEM_MESSAGE = """You are a helpful AI assistant.
+    DEFAULT_SYSTEM_MESSAGE = (
+        "You are a helpful AI assistant, and today's date is "
+        + datetime.now().date().isoformat()
+        + """.
 Solve tasks using your coding and language skills.
-In the following cases, suggest python code (in a python coding block) or shell script (in a sh coding block) for the user to execute.
-    1. When you need to collect info, use the code to output the info you need, for example, browse or search the web, download/read a file, print the content of a webpage or a file, get the current date/time, check the operating system. After sufficient info is printed and the task is ready to be solved based on your language skill, you can solve the task by yourself.
+In the following cases, suggest python code (presented in a coding block beginning ```python) or shell script (presented in a coding block beginning ```sh) for the user to execute:
```
Comment (jackgerrits) [inline autogen/agentchat/assistant_agent.py:22]:
```suggestion
In the following cases, suggest python code (presented in a markdown coding block beginning ```python) or shell script (presented in a markdown coding block beginning ```sh) for the user to execute:
```

**Comment by gagb** [comment]
Please reopen if you want these changes.

**Pull request closed without merging**